src/Parsers/PRQL/ParserPRQLQuery.cpp

```diff
@@ -0,0 +1,84 @@
+#include <string>
+#include <Parsers/PRQL/ParserPRQLQuery.h>
+
+#include "Parsers/Lexer.h"
+#include "config.h"
+
+#if USE_PRQL
+#    include <prql.h>
+#endif
+
+#include <Parsers/ParserQuery.h>
+#include <Parsers/ParserSetQuery.h>
+#include <Parsers/parseQuery.h>
+#include <base/scope_guard.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int SYNTAX_ERROR;
+    extern const int SUPPORT_IS_DISABLED;
+}
+
+bool ParserPRQLQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserSetQuery set_p;
+
+    if (set_p.parse(pos, node, expected))
+        return true;
+
+#if !USE_PRQL
+    throw Exception(
+        ErrorCodes::SUPPORT_IS_DISABLED, "PRQL is not available. Rust code or PRQL itself may be disabled. Use another dialect!");
+#else
+    const auto * begin = pos->begin;
+
+    while (!pos->isEnd() && pos->type != TokenType::Semicolon)
```

Comment:
Doesn't the client split the queries with the semicolon first? Maybe it is enough to process the whole string here?

Reply:
The issue is the client uses the same parser as the server, thus we have to handle the semicolons to make the client work properly.

Reply:
Oh, you are right, I guess worth to put a comment then.